virtool/users/sessions.py

```diff
@@ -131,42 +137,42 @@ async def create_reset_code(
     """
     reset_code = secrets.token_hex(32)
 
-    await db.sessions.update_one(
-        {"_id": session_id},
-        {
-            "$set": {
-                "reset_code": reset_code,
-                "reset_remember": remember,
-                "reset_user_id": user_id,
-            }
-        },
-    )
+    session = json.loads(await redis.get(session_id))
+    session_expiry = await redis.ttl(session_id)
+
+    session.update({"reset_code": reset_code, "reset_remember": remember, "reset_user_id": user_id})
+    await redis.set(session_id, json.dumps(session, cls=CustomEncoder), expire=session_expiry)
 
     return reset_code
 
 
-async def clear_reset_code(db: virtool.mongo.core.DB, session_id: str):
+async def clear_reset_code(redis: Redis, session_id: str):
     """
     Clear the reset information attached to the session associated with the passed
     `session_id`.
 
-    :param db: the application database client
+    :param redis: the application redis client
     :param session_id: the session id
 
     """
-    await db.sessions.update_one(
-        {"_id": session_id},
-        {"$unset": {"reset_code": "", "reset_remember": "", "reset_user_id": ""}},
-    )
+    session = json.loads(await redis.get(session_id))
+    session_expiry = await redis.ttl(session_id)
+
+    for key in ["reset_code", "reset_remember", "reset_user_id"]:
+        if key in session:
+            del session[key]
```

Comment:
```suggestion
    session = {key: session[key] for key in session if key not in {"reset_code", "reset_remember", "reset_user_id"}}
```
You could go further avoid declaring `session` again and stick this right in the `dumps`  call.

Reply:
Changed to match styles, but I feel compelled to point out that this forces iteration through the whole dict. Since the session dict is so small the performance downgrade of using a dict comprehension should be minimal, but trading O(1) for O(n) hurts me a little.

Reply:
At this scale it will make no difference, however feel free to use your implementation if it hurts you. I would suggest using try-except though instead of `if`.

The payoffs (negligible) of my suggestion are:
* Resistant to `KeyError`.
* Does not mutate the dictionary.

Reply:
Nah I am content to go with you solution. I agree that the performance difference would be negligible and I would rather keep the code consistent given how small the session dict is. If the dict were to really grow (which I don't foresee) then the performance difference could matter but not as it is now.